cocos/ui/button.ts

```diff
@@ -979,6 +990,38 @@ export class Button extends Component {
             this._updateScaleTransition(state);
         }
     }
+
+    protected _xrHoverEnter() {
+        this._onMouseMoveIn();
+        this._updateState();
+    }
+
+    protected _xrHoverExit() {
+        this._onMouseMoveOut();
+        if (this._pressed) {
+            this._pressed = false;
+            this._updateState();
+        }
+    }
+
+    protected _xrClick() {
+        if (!this._interactable || !this.enabledInHierarchy) { return; }
+        this._pressed = true;
+        this._updateState();
+    }
+
+    protected _xrUnClick() {
+        if (!this._interactable || !this.enabledInHierarchy) {
+            return;
+        }
+
+        if (this._pressed) {
+            ComponentEventHandler.emitEvents(this.clickEvents, this);
+            this.node.emit(EventType.CLICK, this);
+        }
+        this._pressed = false;
+        this._updateState();
+    }
```

Comment:
```suggestion
    private _xrHoverEnter() {
        this._onMouseMoveIn();
        this._updateState();
    }

    private _xrHoverExit() {
        this._onMouseMoveOut();
        if (this._pressed) {
            this._pressed = false;
            this._updateState();
        }
    }

    private _xrClick() {
        if (!this._interactable || !this.enabledInHierarchy) { return; }
        this._pressed = true;
        this._updateState();
    }

    private _xrUnClick() {
        if (!this._interactable || !this.enabledInHierarchy) {
            return;
        }

        if (this._pressed) {
            ComponentEventHandler.emitEvents(this.clickEvents, this);
            this.node.emit(EventType.CLICK, this);
        }
        this._pressed = false;
        this._updateState();
    }
```


Suggest to set as private method

Reply:
`protected` means the method is public,
and we need to consider maintenance compatibility later